tests/hw/esp-tester/mini-jag.toit

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2024 Toitware ApS.
+// Use of this source code is governed by a Zero-Clause BSD license that can
+// be found in the tests/LICENSE file.
+
+import crypto.crc
+import esp32
+import net
+import system.containers
+import uuid show Uuid
+
+main:
+  cause := esp32.wakeup-cause
+  print "Wakeup cause: $cause"
+  // It looks like resetting the chip through the UART yields RST-UNKNOWN.
+  if cause == esp32.ESP-RST-POWERON or cause == esp32.ESP-RST-UNKNOWN:
+    install-new-test
+    esp32.deep-sleep Duration.ZERO
+  run-test
+
+install-new-test:
+  // Clear all existing containers.
+  ids := containers.images.map: | image/containers.ContainerImage |
+    image.id
+  ids.do: | id/Uuid |
+    if id != containers.current:
+      containers.uninstall id
+
+  network := net.open
+  server-socket := network.tcp-listen 0
+  print "$network.address:$server-socket.local-address.port"
+  print "MINI-JAG LISTENING"
```

Comment:
Maybe use the constant from shared.toit?

Reply:
done.